livy/src/main/java/org/apache/zeppelin/livy/BaseLivyInterprereter.java

```diff
@@ -229,7 +229,31 @@ private InterpreterResult getResultFromStatementInfo(StatementInfo stmtInfo,
     } else {
       //TODO(zjffdu) support other types of data (like json, image and etc)
       String result = stmtInfo.output.data.plain_text;
-      if (result != null) {
+      Object magicResult = null;
+      
+      // check magic first
+      if ((magicResult = stmtInfo.output.data.application_livy_table_json) != null) {
```

Comment:
This is one sample I see for table format.
```
            "headers": [
              {
                "type": "BIGINT_TYPE",
                "name": "_1"
              },
              {
                "type": "STRING_TYPE",
                "name": "_2"
              }
            ],
            "data": [
              [
                1,
                "a"
              ],
              [
                2,
                "b"
              ]
            ]
```

Could we represent it using POJO so that gson can do deserialization automatically ?
If this is too complicated and not worth to do, then we'd better to wrap the parsing into one function.